src/compiler/parser.ts

```diff
@@ -9287,16 +9287,17 @@ namespace IncrementalParser {
         }
     }
 
-    function moveElementEntirelyPastChangeRange(element: IncrementalElement, isArray: boolean, delta: number, oldText: string, newText: string, aggressiveChecks: boolean) {
-        if (isArray) {
-            visitArray(element as IncrementalNodeArray);
+    function moveElementEntirelyPastChangeRange(element: IncrementalNode | IncrementalNodeArray, delta: number, oldText: string, newText: string, aggressiveChecks: boolean) {
+        if (isArray(element)) {
+            visitArray(element);
         }
         else {
-            visitNode(element as IncrementalNode);
+            visitNode(element);
         }

```

Comment:
I could restore this if desired; it's technically faster to statically say if it's an array or not, but I don't think this is a pattern I've seen in the repo.

Reply:
Restored, instead using explicit type assertions on the functions which better represent the relative unsafety of this pattern.